reporter/tap.js

```diff
@@ -0,0 +1,77 @@
+(function() {
+
+function reporter( rep ) {
+	if ( rep === "tap" ) {
+		tapReporter();
+	}
+};
+
+reporter.print = function( val ) {
+	console.log( val );
+};
+
+QUnit.reporter = reporter;
+
+function tapReporter() {
+	var testCount = 0;
+	var failed = 0;
+
+	QUnit.begin( function() {
+		QUnit.reporter.print( "TAP version 13" );
```

Comment:
It should be specific to the TAP reporter, or even better to a TAP reporter _instance_. Testing code code then boil down to something like `QUnit.reporter( TAPReporterFactory({ print: captureOutput }) )` (modulo construction/configuration/registration bikeshedding).